src/item-manager.js

```diff
@@ -0,0 +1,25 @@
+class ItemManager {
+   constructor() {
+      this.items = [];
```

Comment:
a very cool thing you can do: receive a callback of `onItemChange`, and call it with `this.items` argument each time.
The main script can send `renderItems` to the `ItemManager` as `callback`, and you won't have to call it manually each time! (you won't need to maintain the `render` variable)

Reply:
Awesome! just implemented it and it shortened many line of code and easier total logic. Thanks